lib/querystring.js

```diff
@@ -5,6 +5,18 @@
 const QueryString = exports;
 const Buffer = require('buffer').Buffer;
 
+// This constructor is used to store parsed query string values. Instantiating
+// this is faster than explicitly calling `Object.create(null)` to get a
+// "clean" empty object
+function ParsedQueryString() {}
+ParsedQueryString.prototype = Object.create(null, {
+  constructor: {
```

Comment:
It's not about `instanceof` per-se, it's about `.hasOwnProperty` and other
methods from `Object.prototype` that most devs give, wrongly, for granted
with objects. This won't affect my code but it will break every code with
`for/in` loops. I personally don't care
On Apr 5, 2016 1:59 PM, "Brian White" <notifications@github.com> wrote:

> In lib/querystring.js
> <https://github.com/nodejs/node/pull/6055#discussion_r58532723>:
>
> > @@ -5,6 +5,18 @@
> >  const QueryString = exports;
> >  const Buffer = require('buffer').Buffer;
> >
> > +// This constructor is used to store parsed query string values. Instantiating
> > +// this is faster than explicitly calling `Object.create(null)` to get a
> > +// "clean" empty object
> > +function ParsedQueryString() {}
> > +ParsedQueryString.prototype = Object.create(null, {
> > +  constructor: {
>
> constructor is removed now.
>
> IMHO I think more people are going to be using typeof foo === 'object'
> rather than foo instanceof Object. Perhaps @ChALkeR
> <https://github.com/ChALkeR> can give us some insight about this?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/nodejs/node/pull/6055/files/ef5f84a85820794a572ba0deb7e15a8b1622b9f4#r58532723>
>